Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -81,6 +93,81 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
+                await _scheduleRepository.BulkСancelInterviewsAsync(cancelledInterviews);
+                await _candidateService.BulkUpdateStatusAsync(cancelledInterviews.Select(i => i.CandidateId), projectId, _statusHelper.GetStatusDown(cuurentStatusCadidates));
+            }
+
+            var appointedInterviews = _mapper.Map<IEnumerable<Interview>>(interviews.Where(i => i.IsAppointment == true));
+            var users = await _userService.GetUsersShortByIdsAsync(appointedInterviews.Select(i => i.UserId));
+            var cadidatesIdsAppointedInterviews = appointedInterviews.Select(i => i.CandidateId);
+            cuurentStatusCadidates = allCadidates.Where(c => cadidatesIdsAppointedInterviews.Contains(c.Id)).Select(c => c.ProjectResults.FirstOrDefault(p => p.ProjectId == projectId).Status).FirstOrDefault();
+
+            var usersForInvite = users.Select(c => new UserEmailInfo() { Id = c.Id, Email = c.Email, Name = c.Name }).ToList();
+            var candidatesForInvite = allCadidates.Where(c => cadidatesIdsAppointedInterviews.Contains(c.Id))
+                                        .Select(c => new CandidateEmailInfo()
+                                        {
+                                            Email = c.Email,
+                                            Id = c.Id,
+                                            Name = c.Name,
+                                            PhoneNumber = c.PhoneNumber,
+                                            Skype = c?.Contacts?.FirstOrDefault(s => s.Type == "Skype")?.Value,
+                                        }).ToList();
+
+            var args = new AssignedInterviewEventArgs()
+            {
+                Interviews = appointedInterviews
+                                .Select(i => new InterviewEmailInfo()
+                                {
+                                    AppointDateTimeUtc = i.AppointDateTimeUtc,
+                                    InterviewType = (InterviewType)Enum.Parse(typeof(InterviewType), cuurentStatusCadidates.ToString(), true),
+                                    Candidate = candidatesForInvite.FirstOrDefault(c => c.Id == i.CandidateId),
+                                    User = usersForInvite.FirstOrDefault(u => u.Id == i.UserId),
+                                }),
+            };
+
+            if (appointedInterviews.Any())
+            {
+                await _scheduleRepository.BulkAppointInterviewsAsync(
+                    appointedInterviews,
+                    allCadidates.Where(c => cadidatesIdsAppointedInterviews.Contains(c.Id)).Select(c => new ScheduleCandidateInfo()
+                    {
+                         BestTimeToConnect = c.BestTimeToConnect,
+                         Id = c.Id,
+                         Name = c.Name,
+                         Surname = c.Surname,
+                         Email = c.Email,
+                         Skype = c?.Contacts?.FirstOrDefault(s => s.Type == "Skype")?.Value,
```

Comment:
why you write c?, if in prev lines you already wrote c.Email for example, if it woulf null exception, ? won't save you on last line

Reply:
+